api/v1beta1/conditions_consts.go

```diff
@@ -38,6 +38,8 @@ const (
 	VMDeletingReason = "VMDeleting"
 	// VMProvisionFailedReason used for failures during vm provisioning.
 	VMProvisionFailedReason = "VMProvisionFailed"
+	// VMUnhealthyReason used for failures when the VM is in an unhealthy state.
+	VMUnhealthyReason = "VMUnhealthy"
```

Comment:
Yes that's the reason! At first, I created a VMUnhealthy condition, but realized that it would be redundant with the VMRunning condition, since when the VM is unhealthy, we would want to replace it with another machine anyways. In order to do that we would need to set the VMRunning condition to false. I think the general rule is that if the reason doesn't fall under a pre-existing condition, we would want to create a new one. @CecileRobertMichon would you be able to confirm that my understanding is correct?

On a separate note, there is another rule with conditions saying that all of them must be positive, meaning VMUnhealthy would break that rule. If we did want to add another condition, we would name it VMHealthy instead.

Reply:
> I think the general rule is that if the reason doesn't fall under a pre-existing condition, we would want to create a new one.

That's correct.

In this case though, I'm leaning towards adding a new condition since the user-assigned identity doesn't really affect the VM running state (the user would still see the VM as `Running` in the Azure Portal). I also wouldn't call the VM "unhealthy", but maybe more something specific like `VMIdentitiesReady` with a reason `UserAssignedIdentityMissing` or something like that.